### PR TITLE
Gecode fixes

### PR DESCRIPTION
**Pull request by liuq**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

The SHA256 signature of the source tar.gz file was wrong and it has been fixed.
Moreover, a new option "with-qt" has been added to allow the compilation of
the Qt based graphical search inspector "gist".

### CODE REVIEW

**Comment by DomT4** [comment]
> The SHA256 signature of the source tar.gz file was wrong and it has been fixed.

Can you find out why it changed? Thanks. It will have been correct whenever the change was merged.

> Moreover, a new option "with-qt" has been added to allow the compilation of
the Qt based graphical search inspector "gist".

Can it use Qt5? We'd prefer not to add any more `qt` dependencies because it's very likely `qt` won't work on Sierra & we aren't intending to carry another hacky patch for another year.

**Comment by MikeMcQuaid** [comment]
@liuq Any news on this?

**Comment by liuq** [comment]
I cannot deploy before testing with QT5 (as requested), but homebrew QT5 is currently broken.

**Comment by ilovezfs** [comment]
@liuq https://github.com/Homebrew/homebrew-core/pull/5619 is sufficient to build if you have the patience for a qt5 build.

**Comment by MikeMcQuaid** [comment]
We won't be adding any more `qt` dependencies so happy to make this PR Qt-less for now.

**Comment by liuq** [comment]
@MikeMcQuaid In general I would agree, however the value of Gecode w.r.t the rsystems is in its (qt-based) debugging tools. I try to check with the #5619 if I can get a working qt5 environment.

**Comment by MikeMcQuaid** [comment]
> I try to check with the #5619 if I can get a working qt5 environment.

Thanks!

**Comment by liuq** [comment]
That's it, the current formula is qt5 compliant.

**Comment by Moisan** [comment]
I wrote [upstream](http://www.gecode.org/bugs.html) to ask if the change to the source without version change was intentional.

**Comment by liuq** [comment]
Yes, it is, the version is the very same but, probably, the source file on
the repository has changed (and the SHA signature accordingly).

Il giorno lun 10 ott 2016 alle 16:23 Thierry Moisan <
notifications@github.com> ha scritto:

**Comment by MikeMcQuaid** [comment]
I'd rather we wait for a response first to ensure they've not been hacked.

**Comment by liuq** [comment]
@MikeMcQuaid will you handle that? In case you can write to Christian Schulte cschulte@kth.se, who is one of the main gecode maintainers

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by Moisan** [comment]
I've got a [response from upstream](http://www.gecode.org/pipermail/bugs/2016-October/000336.html). 

> this was not intentional, but rather a side effect of updating one of the binary packages (which caused a new version of the source packages to be uploaded as well).  There has not been any change to the source code itself so it is safe to update the SHA256 to the new one.

**Comment by MikeMcQuaid** [comment]
@Moisan Thanks!

File: Formula/gecode.rb
Comment (MikeMcQuaid) [inline Formula/gecode.rb]: Instead use `ENV.cxx11` and see if that's sufficient.

File: Formula/gecode.rb
Comment (MikeMcQuaid) [inline Formula/gecode.rb]: Any chance you could add a simple test e.g. ideally something more substantial than e.g. `--version` or `--help` but just use them if necesssary? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/gecode.rb
Comment (liuq) [inline Formula/gecode.rb]: Done!

File: Formula/gecode.rb
Comment (MikeMcQuaid) [inline Formula/gecode.rb]: `-I#{include}`

File: Formula/gecode.rb
Comment (MikeMcQuaid) [inline Formula/gecode.rb]: Just hard-code these args (but split them over multiple lines to avoid exceeding 80 characters).

File: Formula/gecode.rb
Comment (MikeMcQuaid) [inline Formula/gecode.rb]: Replace the above two lines with `assert_match "{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}", shell_output("./test")`

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/gecode.rb
Comment (liuq) [inline Formula/gecode.rb]: Ok, did it!

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by liuq** [comment]
It's a pleasure to contribute to the community.

Il giorno mer 12 ott 2016 alle 18:16 Mike McQuaid <notifications@github.com>
ha scritto:

**Pull request closed without merging**
